src/domain/relay/contracts/safe-contract.helper.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { faker } from '@faker-js/faker';
+import { concatHex, encodeFunctionData, getAddress, pad, toHex } from 'viem';
+import { SafeContractHelper } from '@/domain/relay/contracts/safe-contract.helper';
+import { CALL_OPERATION } from '@/domain/safe/entities/operation.entity';
+
+const ABI_ITEM = {
+  inputs: [
+    {
+      internalType: 'address',
+      name: 'to',
+      type: 'address',
+    },
+    {
+      internalType: 'uint256',
+      name: 'value',
+      type: 'uint256',
+    },
+    {
+      internalType: 'bytes',
+      name: 'data',
+      type: 'bytes',
+    },
+    {
+      internalType: 'enum Enum.Operation',
+      name: 'operation',
+      type: 'uint8',
+    },
+    {
+      internalType: 'uint256',
+      name: 'safeTxGas',
+      type: 'uint256',
+    },
+    {
+      internalType: 'uint256',
+      name: 'baseGas',
+      type: 'uint256',
+    },
+    {
+      internalType: 'uint256',
+      name: 'gasPrice',
+      type: 'uint256',
+    },
+    {
+      internalType: 'address',
+      name: 'gasToken',
+      type: 'address',
+    },
+    {
+      internalType: 'address payable',
+      name: 'refundReceiver',
+      type: 'address',
+    },
+    {
+      internalType: 'bytes',
+      name: 'signatures',
+      type: 'bytes',
+    },
+  ],
+  name: 'execTransaction',
+  outputs: [
+    {
+      internalType: 'bool',
+      name: 'success',
+      type: 'bool',
+    },
+  ],
+  stateMutability: 'payable',
+  type: 'function',
+};
+
+describe('Safe Contract Helper Tests', () => {
+  let target: SafeContractHelper;
+
+  beforeEach(() => {
+    target = new SafeContractHelper();
+  });
+
+  it('decodes an execTransaction correctly', () => {
+    const to = getAddress(faker.finance.ethereumAddress());
+    const value = faker.number.bigInt();
+    const callData = getExecTransactionCallData({ to, value });
+
+    const actual = target.decode(SafeContractHelper.EXEC_TRANSACTION, callData);
+
+    expect(actual).toEqual({ data: '0x00', to: to, value: value });
+  });
+
+  it('decoding a non execTransaction call throws', () => {
+    const functionSignature = faker.string.hexadecimal({
+      length: 8,
+    }) as `0x${string}`;
+    const arg1 = pad(getAddress(faker.finance.ethereumAddress()));
+    const arg2 = pad(getAddress(faker.finance.ethereumAddress()));
+    const callData = concatHex([functionSignature, arg1, arg2]);
+
+    expect(() => {
+      target.decode(SafeContractHelper.EXEC_TRANSACTION, callData);
+    }).toThrow();
+  });
+
+  it('isCall returns true for a execTransaction call', () => {
+    const callData = getExecTransactionCallData();
+
+    const actual = target.isCall(callData);
+
+    expect(actual).toBe(true);
+  });
```

Comment:
I think we should add coverage for Safe contract calls as well.

Reply:
Done in [831f2e9](https://github.com/safe-global/safe-client-gateway/pull/680/commits/831f2e9947aaecddfaff2fb97d987b3378f36ba3).

It does not cover all the Safe Contract functions but the more commonly used ones though.